cedar-policy-validator/src/err.rs

```diff
@@ -26,78 +26,82 @@ use thiserror::Error;
     /// Cycle in the schema's action hierarchy.
-    #[error("Cycle in action hierarchy")]
+    #[error("cycle in action hierarchy")]
     CycleInActionHierarchy,
     /// Parse errors occurring while parsing an entity type.
-    #[error("Parse error in entity type: {}", Self::format_parse_errs(.0))]
+    #[error("parse error in entity type: {}", Self::format_parse_errs(.0))]
     EntityTypeParseError(ParseErrors),
     /// Parse errors occurring while parsing a namespace identifier.
-    #[error("Parse error in namespace identifier: {}", Self::format_parse_errs(.0))]
+    #[error("parse error in namespace identifier: {}", Self::format_parse_errs(.0))]
     NamespaceParseError(ParseErrors),
     /// Parse errors occurring while parsing an extension type.
-    #[error("Parse error in extension type: {}", Self::format_parse_errs(.0))]
+    #[error("parse error in extension type: {}", Self::format_parse_errs(.0))]
     ExtensionTypeParseError(ParseErrors),
     /// Parse errors occurring while parsing the name of one of reusable
     /// declared types.
-    #[error("Parse error in common type identifier: {}", Self::format_parse_errs(.0))]
+    #[error("parse error in common type identifier: {}", Self::format_parse_errs(.0))]
     CommonTypeParseError(ParseErrors),
     /// The schema file included an entity type `Action` in the entity type
     /// list. The `Action` entity type is always implicitly declared, and it
     /// cannot currently have attributes or be in any groups, so there is no
     /// purposes in adding an explicit entry.
-    #[error("Entity type `Action` declared in `entityTypes` list")]
+    #[error("entity type `Action` declared in `entityTypes` list")]
     ActionEntityTypeDeclared,
     /// One or more action entities are declared with `attributes`, but this is
     /// not currently supported.
-    #[error("Actions declared with `attributes`: [{}]", .0.iter().join(", "))]
-    ActionEntityAttributes(Vec<String>),
+    #[error("actions declared with `attribute(s)`: [{}]", .0.iter().join(", "))]
+    ActionHasAttributes(Vec<String>),
+    /// `context` or `shape` fields are not records
     #[error("{0} is declared with a type other than `Record`")]
     ContextOrShapeNotRecord(ContextOrShape),
-    /// An Action Entity (transitively) has an attribute that is an empty set
-    #[error("An action entity has an attribute that is an empty set")]
-    ActionEntityAttributeEmptySet,
-    /// An Action Entity (transitively) has an attribute of unsupported type (ExprEscape, EntityEscape or ExtnEscape)
-    #[error("An action entity has attribute with unsupported type: (escaped expression, entity or extension)")]
-    ActionEntityAttributeUnsupportedType,
+    /// An action entity (transitively) has an attribute that is an empty set
+    /// This error variant should be used when `PermitAttributes` is enabled.
+    #[error("action has an attribute that is an empty set")]
+    ActionAttributesContainEmptySet,
```

Comment:
would be great to specify the name of the action that's in violation

Reply:
Good idea. Done.